### PR TITLE
Allow @ in filenames and valid UTF letters in usernames

### PR DESCRIPTION
**Pull request by zekoz**
Testing plan:
* Try to use fancy usernames with any characters. Letters and numbers should be allowed, punctuation etc should not. Things like chat should still behave as expected.
* Try to copy a ruleset as `ruleset@abcd` and load it. Should work.

### CODE REVIEW

**Comment by lmoureaux** [comment]
What's going on with the CI?

File: common/player.cpp [hunk 1]
```diff
@@ -1796,8 +1797,30 @@ int player_in_territory(const struct player *pplayer,
  */
 bool is_valid_username(const char *name)
 {
-  return (strlen(name) > 0 && !QChar::isDigit(name[0]) && is_ascii_name(name)
-          && fc_strcasecmp(name, ANON_USER_NAME) != 0);
+  QString username(name);
+
+  if (username.length() == 0) {
```
Comment (lmoureaux) [inline common/player.cpp:1802]: `isEmpty()`

File: utility/support.cpp [hunk 2]
```diff
@@ -501,6 +501,33 @@ size_t fc_strlcpy(char *dest, const char *src, size_t n)
   }
 }
 
+size_t fc_strlcpy(char *dest, const QStringView &src, size_t n)
```
Comment (lmoureaux) [inline utility/support.cpp:504]: It would make sense to call this function from the `const char *` overload above

File: common/player.h [hunk 3]
```diff
@@ -28,7 +28,7 @@
 /* If changing this, be sure to adjust loading of pre-2.6 savegames
  * which depend on saved username to tell if user (or ranked_user) is
  * anonymous. */
-#define ANON_USER_NAME N_("Unassigned")
+const QString kAnonUserName("Unassigned");
```
Comment (lmoureaux) [inline common/player.h:31]: Use `QStringLiteral` to avoid an allocation before even `main` is called.

**Review by lmoureaux — COMMENTED** [review]
I like the conversion of a preprocessor macro to a `QString` and the overloading of `fc_strlcpy`. Few code nitpicks.

I think the errors spotted in the CI are easy to fix.

**Comment by lmoureaux** [comment]
Made the patch simpler to avoid `QString` <-> `char *` issues. Should be good to go now.

**Review by jwrober — APPROVED** [review]

**Pull request merged**